### DISCUSSION TITLE
Use in a kubernetes microservice as an application scheduler

### DISCUSSION THREAD

**Discussion by altergs** [quartznet/quartznet#1542] [Q&A]
Hi,
what are the best practices if I want to use Quartz.NET in a kubernetes microservice (given that a service may have multiple instances, etc).  How do I make sure that only one instances processes a job, for example?

And is it a good use case for quartz.net to be used as an application-level scheduler in a microservice application, so I can schedule recurring calls to my microservices (e.g. call a specific REST method on a specific microservice once a week).

I'd be using it with AWS RDS as a backing store.

**Comment by lahma** [comment]
Disclaimer, I have never done neither k8s nor AWS, only Azure in this equation. 

If your container orchestrator doesn't have built-in scheduler that would handle this for you, Quartz.NET might be a good choice. If you go with Quartz.NET, it's at least portable between cloud providers if the containers etc stay the same.

As long as you have database-backed job store, it should handle concurrency and distribution for you. You should just follow clustered scheduler setup. AWS RDS isn't supported by default though, *if* it works with default MySQL providers you should be golden though.
